docs/en/02_Developer_Guides/14_Files/03_File_Security.md

```diff
@@ -352,11 +359,70 @@ SilverStripe\Filesystem\Flysystem\FlysystemAssetStore:
     Pragma: 'no-cache'
 ```
 
+### Configuring file HTTP response code
+
+When a user tries to access a file that exists, but for which they do not have access,
+Silverstripe CMS will return a "404 Not found" response rather than a "403 Denied" to
+avoid revealing the existence of the file.
+
+You can customise the response codes for various types of request via
+configuration flags on `FlysystemAssetStore`.
+
+```yml
+# Add this code snippet to your YML configuration
+SilverStripe\Filesystem\Flysystem\FlysystemAssetStore:
+  denied_response_code: 403 # This default to 404
+  missing_response_code: 404
+  redirect_response_code: 302
+  permanent_redirect_response_code: 301
+```
+
+### Updating a file HTTP response before it's sent back to the browser
+
+`silverstripe/assets` 1.6 and above allows you to intercept the file HTTP response
+before it's sent to the client by applying an `Extension` to `FlysystemAssetStore`.
+
+To achieve this create an `Extension` and implement the `updateResponse` method.
+
+```php
+<?php
+
+namespace App\MySite;
+
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Extension;
+
+class AssetStoreExtension extends Extension
+{
+    /**
+     * @param HTTPResponse $response Update this object to modify the response
+     * @param string $asset   Path of the request minus the `assets` prefix
+     * @param array $context  This array contains some resolution information from
+     *                        FlysystemAssetStore. It may be empty. It may contain
+     *                        a `visibility` key to say if we are serving a public
+     *                        or protected file. It may contain a `parsedFileID`
+     *                        detailing how FlysystemAssetStore has resolved $asset
+     */
+    public function updateResponse($response, $asset, $context)
```

Comment:
```suggestion
    public function updateResponse(HTTPResponse $response, string $asset, array $context): void
```

I think we should start documenting with PHP 7 syntax where possible

Reply:
Fixed.